src/dft/descriptor.cxx

```diff
@@ -26,51 +27,109 @@ namespace detail {
 
 template <precision prec, domain dom>
 void descriptor<prec, dom>::set_value(config_param param, ...) {
-    int err = 0;
+    if (pimpl_) {
+        throw mkl::invalid_argument("DFT", "set_value",
+                                    "Cannot set value on committed descriptor.");
+    }
     va_list vl;
     va_start(vl, param);
     switch (param) {
-        case config_param::INPUT_STRIDES: [[fallthrough]];
+        case config_param::FORWARD_DOMAIN:
+            throw mkl::invalid_argument("DFT", "set_value", "Read-only parameter.");
+            break;
+        case config_param::DIMENSION:
+            throw mkl::invalid_argument("DFT", "set_value", "Read-only parameter.");
+            break;
+        case config_param::LENGTHS: {
+            if (values_.rank == 1) {
+                values_.dimensions = std::vector<std::int64_t>{ va_arg(vl, std::int64_t) };
+            }
+            else {
+                auto ptr = va_arg(vl, std::int64_t*);
+                if (ptr == nullptr) {
+                    throw mkl::invalid_argument("DFT", "set_value", "config_param is nullptr.");
+                }
+                std::copy(ptr, ptr + values_.rank + 1, values_.dimensions.begin());
```

Comment:
This seems to be out of bounds. I think the correct code should be:

std::copy(ptr, ptr + values_.rank, values_.dimensions.begin());

Reply:
I agree. Fixed here and other similar locations in [ed09ed2](https://github.com/hjabird/oneMKL/pull/3/commits/ed09ed20bbb0905584c348a1bb06ef11059933be).